python/fledge/common/acl_manager.py

```diff
@@ -27,27 +38,44 @@ async def _notify_service_about_acl_change(self, entity_name, acl, reason):
+            _logger.info("Moved {} to pending. And pending is {}".format(entity_name,
+                                                                         self._pending_notifications))
+
+            self._pending_notifications[entity_name] = acl
             return
         else:
-            _logger.info("Notifying the service"
-                         " {} about {}".format(entity_name, reason))
-            from fledge.common.microservice_management_client.microservice_management_client import MicroserviceManagementClient
-            _logger.info("The host and port is {} {} and "
-                         "entity is {}".format(service._address,
-                                               service._management_port,
-                                               entity_name))
-            mgt_client = MicroserviceManagementClient(service._address,
-                                                      service._management_port)
-            _logger.info("Connect established with {} at {} and port {}".format(entity_name,
-                                                                                service._address,
-                                                                                service._management_port))
-            await mgt_client.update_service_for_acl_change_security(acl=acl,
-                                                                    reason=reason)
-            _logger.info("Notified the {} about {}".format(entity_name, reason))
+            try:
+                _logger.info("Notifying the service"
+                             " {} about {}".format(entity_name, reason))
+                from fledge.common.microservice_management_client.microservice_management_client import MicroserviceManagementClient
+                _logger.info("The host and port is {} {} and "
+                             "entity is {}".format(service._address,
+                                                   service._management_port,
+                                                   entity_name))
+                mgt_client = MicroserviceManagementClient(service._address,
+                                                          service._management_port)
+                _logger.info("Connect established with {} at {} and port {}".format(entity_name,
+                                                                                    service._address,
+                                                                                    service._management_port))
+                await mgt_client.update_service_for_acl_change_security(acl=acl,
+                                                                        reason=reason)
+                _logger.info("Notified the {} about {}".format(entity_name, reason))
+                # clearing the pending notifications if any.
+                if entity_name in self._pending_notifications:
```

Comment:
only Unresponsive state notifications should be in this in pending list or Failed even when running